android/core/src/test/java/com/pega/vehicledamagemodeling/api/SelectedPartsTest.java

```diff
@@ -0,0 +1,179 @@
+        SelectedPartsRepository selectedPartsRepository = new SelectedPartsRepository();
+        selectedPartsRepository.add("roof",new Material());
+        selectedPartsRepository.add("front bumper",new Material());
+
+        HashSet<String> expected = new HashSet<>();
+        expected.add("roof");
+        expected.add("front bumper");
+
+        //when
+        HashSet<String> result = selectedPartsRepository.getSelectedParts();
+
+        //then
+        assertEquals(expected,result);
+    }
+
+    @Test
+    public void whenSelectionIsEmptyThenReturnEmptyHashSet(){
+        //given
+        SelectedPartsRepository selectedPartsRepository = new SelectedPartsRepository();
+        HashSet<String> expected = new HashSet<>();
+
+        //when
+        HashSet<String> result = selectedPartsRepository.getSelectedParts();
+
+        //then
+        assertEquals(expected,result);
+    }
+
+    @Test
+    public void whenJsonIsEmptyThenReturnEmptyJson(){
+        //given
+        JsonObject initJson = new JsonObject();
+        SelectedPartsRepository selectedPartsRepository = new SelectedPartsRepository();
+        selectedPartsRepository.setInitJson(initJson);
+
+        //when
+        JsonObject result = selectedPartsRepository.getInitJson();
+
+        //then
+        assertEquals(initJson,result);
+    }
+
+    @Test
+    public void whenJsonIsNotEmptyThenReturnCorrectJson(){
+        //given
+        JsonObject initJson = new JsonObject();
+        JsonArray partsArray = new JsonArray();
+        JsonObject jsonProperty = new JsonObject();
+        jsonProperty.addProperty(Id,"roof");
+        partsArray.add(jsonProperty);
+        initJson.add(Selection,partsArray);
+
+        SelectedPartsRepository selectedPartsRepository = new SelectedPartsRepository();
+        selectedPartsRepository.setInitJson(initJson);
+
+        //when
+        JsonObject result = selectedPartsRepository.getInitJson();
+
+        //then
+        assertEquals(initJson,result);
+    }
+
+    @Test
+    public void whenTextIsNotEmptyThenReturnCorrectText(){
+        //given
+        SelectedPartsRepository selectedPartsRepository = new SelectedPartsRepository();
+        selectedPartsRepository.setMainScreenText("The text");
+        String expected = "The text";
+
+        //when
+        String result = selectedPartsRepository.getMainScreenText();
+
+        //then
+        assertEquals(expected, result);
+    }
+
+    @Test
+    public void whenTextIsEmptyThenReturnEmptyText(){
+        //given
+        SelectedPartsRepository selectedPartsRepository = new SelectedPartsRepository();
+        selectedPartsRepository.setMainScreenText("");
+
+        //when
+        String result = selectedPartsRepository.getMainScreenText();
+
+        //then
+        assertEquals("",result);
+    }
+
+    @Test
+    public void whenPartExistThenDoNotAdd(){
+        //given
+        SelectedPartsRepository selectedPartsRepository = new SelectedPartsRepository();
```

Comment:
Move such things like SelectedPartsRepository selectedPartsRepository = new SelectedPartsRepository(); to setUp or prepare method (I don't know what's the naming convention)